### PR TITLE
NIMBUS-116 Created Dropdown in Table component

### PR DESCRIPTION
**Pull request by purnamashetty**
# Description
Modified the collectionParams structure in the grid param.
Updated the param-state.ts, Table.component.ts, Table.component.html, tree-grid.component.ts and ViewConfig.java.

# Type of Change
- [ ] New feature

### CODE REVIEW

**Comment by openanthem-admin** [comment]
You did it @purnamashetty!

Thank you for signing the Anthem Opensource Software CLA.

File: nimbus-ui/nimbusui/src/app/shared/param-state.ts [hunk 1]
```diff
@@ -336,7 +336,7 @@ export class StyleState implements Serializable<StyleState, string> {
  */
 export abstract class TableBasedData {
 
-    collectionParams: Param[] = [];
+    collectionParams: any[] = [];
```
Comment (lopezton-anthem) [inline nimbus-ui/nimbusui/src/app/shared/param-state.ts]: This should be:

`collectionParams: { [id: string] : Param[]; } = {}`

File: nimbus-ui/nimbusui/src/app/shared/param-state.ts [hunk 2]
```diff
@@ -396,26 +396,26 @@ export abstract class TableBasedData {
...
+    abstract buildRowData(colElemParam: Param, baseParam: Param): RowData;
 }
 
 /**
  * \@author Tony Lopez
  */
 export abstract class RowData {
     values: any = undefined;
-    nestedParams: Param[] = [];
+    nestedParams: any[] = [];
```
Comment (lopezton-anthem) [inline nimbus-ui/nimbusui/src/app/shared/param-state.ts]: This should stay `Param[]`.

File: nimbus-ui/nimbusui/src/app/shared/param-state.ts [hunk 3]
```diff
@@ -463,32 +463,46 @@ export abstract class RowData {
...
+        for (let i = 0; i < this.nestedParams.length; i++){
+            if (this.nestedParams[i].elemId === index){
+                row = this.nestedParams[i];
+            }
+        }
+        if(!row) {
+            row = {elemId: index};
+            this.nestedParams.push(row);
+        }
+        row [param.config.code] = param;
```
Comment (lopezton-anthem) [inline nimbus-ui/nimbusui/src/app/shared/param-state.ts]: We need to discuss this code.

File: nimbus-ui/nimbusui/src/app/shared/param-state.ts [hunk 4]
```diff
@@ -497,10 +511,12 @@ export class TableData extends TableBasedData {
...
 export class TableRowData extends RowData {
 
-    shouldCollect(param: Param): boolean {
+    shouldCollect(param: Param, baseParam?: Param): boolean {
+
         return param.config.uiStyles && 
             (TableComponentConstants.allowedColumnStylesAlias.includes(param.config.uiStyles.attributes.alias)
-                || param.config.uiStyles.attributes.showAsLink);   
+                || param.config.uiStyles.attributes.showAsLink 
+                || baseParam.config.uiStyles.attributes.editRow);
```
Comment (lopezton-anthem) [inline nimbus-ui/nimbusui/src/app/shared/param-state.ts]: `baseParam` is listed as optional, but is mandatory.

File: nimbus-ui/nimbusui/src/app/shared/param-state.ts [hunk 5]
```diff
@@ -26,6 +26,9 @@ import { ViewComponent } from './param-annotations.enum';
 import { DataGroup } from './../components/platform/charts/chartdata';
 import { TableComponentConstants } from './../components/platform/grid/table.component.constants';
 
+export interface nestedParams {[id:string]: Param;};
```
Comment (lopezton-anthem) [inline nimbus-ui/nimbusui/src/app/shared/param-state.ts]: Please rename to `NestedParams`

**Pull request merged**